### ISSUE TITLE
Old version of wheel fails to build PyYAML on Mac Python 3.7

### ISSUE THREAD

**Issue by xerebz** [aws/aws-cli#4243]
If I follow the instructions for building (pip install -r requirements.txt; pip install -e .), I get the following error on Mac using Python 3.7.3. The "assert tag == supported_tags[0]" is miscalculating the ABI (cpdarwin instead of cp-37m-darwin) which was fixed in the next version of wheel (0.25.0). I see that you're still trying to support Python 2.6 so the requirements.txt should be set to something like wheel>=0.25.0,<0.30.0 

```
pip3 install --force-reinstall --no-cache-dir pyyaml

Collecting pyyaml
  Downloading https://files.pythonhosted.org/packages/a3/65/837fefac7475963d1eccf4aa684c23b95aa6c1d033a2c5965ccb11e22623/PyYAML-5.1.1.tar.gz (274kB)
     |████████████████████████████████| 276kB 1.2MB/s
Building wheels for collected packages: pyyaml
  Building wheel for pyyaml (setup.py) ... error
  ERROR: Complete output from command /Users/alaileon/.local/share/virtualenvs/aws-cli-51GcqfER/bin/python3.7 -u -c 'import setuptools, tokenize;__file__='"'"'/private/var/folders/72/cxwx5fy11yg630rtc9tpwsqmw9lqlm/T/pip-install-swpywuf4/pyyaml/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/var/folders/72/cxwx5fy11yg630rtc9tpwsqmw9lqlm/T/pip-wheel-ixq5fth5 --python-tag cp37:
  ERROR: running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-10.13-x86_64-3.7
  creating build/lib.macosx-10.13-x86_64-3.7/yaml
  copying lib3/yaml/scanner.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
  copying lib3/yaml/error.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
  copying lib3/yaml/constructor.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
  copying lib3/yaml/composer.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
  copying lib3/yaml/events.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
  copying lib3/yaml/__init__.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
  copying lib3/yaml/representer.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
  copying lib3/yaml/tokens.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
  copying lib3/yaml/dumper.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
  copying lib3/yaml/cyaml.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
  copying lib3/yaml/parser.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
  copying lib3/yaml/reader.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
  copying lib3/yaml/loader.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
  copying lib3/yaml/resolver.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
  copying lib3/yaml/serializer.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
  copying lib3/yaml/nodes.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
  copying lib3/yaml/emitter.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
  warning: build_py: byte-compiling is disabled, skipping.

  running build_ext
  building '_yaml' extension
  creating build/temp.macosx-10.13-x86_64-3.7
  creating build/temp.macosx-10.13-x86_64-3.7/ext
  clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/Tk.framework/Versions/8.5/Headers -I/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/include/python3.7m -c ext/_yaml.c -o build/temp.macosx-10.13-x86_64-3.7/ext/_yaml.o
  In file included from ext/_yaml.c:596:
  ext/_yaml.h:2:10: fatal error: 'yaml.h' file not found
  #include <yaml.h>
           ^~~~~~~~
  1 error generated.
  Error compiling module, falling back to pure Python
  installing to build/bdist.macosx-10.13-x86_64/wheel
  running install
  running install_lib
  creating build/bdist.macosx-10.13-x86_64
  creating build/bdist.macosx-10.13-x86_64/wheel
  creating build/bdist.macosx-10.13-x86_64/wheel/yaml
  copying build/lib.macosx-10.13-x86_64-3.7/yaml/scanner.py -> build/bdist.macosx-10.13-x86_64/wheel/yaml
  copying build/lib.macosx-10.13-x86_64-3.7/yaml/error.py -> build/bdist.macosx-10.13-x86_64/wheel/yaml
  copying build/lib.macosx-10.13-x86_64-3.7/yaml/constructor.py -> build/bdist.macosx-10.13-x86_64/wheel/yaml
  copying build/lib.macosx-10.13-x86_64-3.7/yaml/composer.py -> build/bdist.macosx-10.13-x86_64/wheel/yaml
  copying build/lib.macosx-10.13-x86_64-3.7/yaml/events.py -> build/bdist.macosx-10.13-x86_64/wheel/yaml
  copying build/lib.macosx-10.13-x86_64-3.7/yaml/__init__.py -> build/bdist.macosx-10.13-x86_64/wheel/yaml
  copying build/lib.macosx-10.13-x86_64-3.7/yaml/representer.py -> build/bdist.macosx-10.13-x86_64/wheel/yaml
  copying build/lib.macosx-10.13-x86_64-3.7/yaml/tokens.py -> build/bdist.macosx-10.13-x86_64/wheel/yaml
  copying build/lib.macosx-10.13-x86_64-3.7/yaml/dumper.py -> build/bdist.macosx-10.13-x86_64/wheel/yaml
  copying build/lib.macosx-10.13-x86_64-3.7/yaml/cyaml.py -> build/bdist.macosx-10.13-x86_64/wheel/yaml
  copying build/lib.macosx-10.13-x86_64-3.7/yaml/parser.py -> build/bdist.macosx-10.13-x86_64/wheel/yaml
  copying build/lib.macosx-10.13-x86_64-3.7/yaml/reader.py -> build/bdist.macosx-10.13-x86_64/wheel/yaml
  copying build/lib.macosx-10.13-x86_64-3.7/yaml/loader.py -> build/bdist.macosx-10.13-x86_64/wheel/yaml
  copying build/lib.macosx-10.13-x86_64-3.7/yaml/resolver.py -> build/bdist.macosx-10.13-x86_64/wheel/yaml
  copying build/lib.macosx-10.13-x86_64-3.7/yaml/serializer.py -> build/bdist.macosx-10.13-x86_64/wheel/yaml
  copying build/lib.macosx-10.13-x86_64-3.7/yaml/nodes.py -> build/bdist.macosx-10.13-x86_64/wheel/yaml
  copying build/lib.macosx-10.13-x86_64-3.7/yaml/emitter.py -> build/bdist.macosx-10.13-x86_64/wheel/yaml
  warning: install_lib: byte-compiling is disabled, skipping.

  running install_egg_info
  running egg_info
  creating lib3/PyYAML.egg-info
  writing lib3/PyYAML.egg-info/PKG-INFO
  writing dependency_links to lib3/PyYAML.egg-info/dependency_links.txt
  writing top-level names to lib3/PyYAML.egg-info/top_level.txt
  writing manifest file 'lib3/PyYAML.egg-info/SOURCES.txt'
  reading manifest file 'lib3/PyYAML.egg-info/SOURCES.txt'
  writing manifest file 'lib3/PyYAML.egg-info/SOURCES.txt'
  Copying lib3/PyYAML.egg-info to build/bdist.macosx-10.13-x86_64/wheel/PyYAML-5.1.1-py3.7.egg-info
  running install_scripts
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/private/var/folders/72/cxwx5fy11yg630rtc9tpwsqmw9lqlm/T/pip-install-swpywuf4/pyyaml/setup.py", line 314, in <module>
      python_requires='>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*',
    File "/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/distutils/core.py", line 148, in setup
      dist.run_commands()
    File "/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/distutils/dist.py", line 966, in run_commands
      self.run_command(cmd)
    File "/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/Users/alaileon/.local/share/virtualenvs/aws-cli-51GcqfER/lib/python3.7/site-packages/wheel/bdist_wheel.py", line 213, in run
      archive_basename = self.get_archive_basename()
    File "/Users/alaileon/.local/share/virtualenvs/aws-cli-51GcqfER/lib/python3.7/site-packages/wheel/bdist_wheel.py", line 161, in get_archive_basename
      impl_tag, abi_tag, plat_tag = self.get_tag()
    File "/Users/alaileon/.local/share/virtualenvs/aws-cli-51GcqfER/lib/python3.7/site-packages/wheel/bdist_wheel.py", line 155, in get_tag
      assert tag == supported_tags[0]
  AssertionError
  ----------------------------------------
  ERROR: Failed building wheel for pyyaml
  Running setup.py clean for pyyaml
Failed to build pyyaml
Installing collected packages: pyyaml
  Found existing installation: PyYAML 5.1.1
    Uninstalling PyYAML-5.1.1:
      Successfully uninstalled PyYAML-5.1.1
  Running setup.py install for pyyaml ... done
Successfully installed pyyaml-5.1.1
```

**Comment by justnance** [comment]
@xerebz - Thanks for posting this issue.  I've reproduced this behavior and confirmed [PyYAML 5.1.1 released on June 6, 2019](https://pypi.org/project/PyYAML/#history) is not compatible with the latest version of the CLI causing the following error:

```
ERROR: Failed building wheel for pyyaml
  Running setup.py clean for pyyaml
Failed to build pyyaml
Installing collected packages: pyyaml
  Found existing installation: PyYAML 5.1.1
    Uninstalling PyYAML-5.1.1:
      Successfully uninstalled PyYAML-5.1.1
  Running setup.py install for pyyaml ... done
Successfully installed pyyaml-5.1.1

```

I'm labeling this issue as enhancement and we are working on getting fixed under PR #4355. 

related to: #4350 #4042

**Comment by romelgomez** [comment]
I think that I facing the same issue in ubuntu. 

```
START: mié dic 25 10:28:02 -05 2019
Collecting awscli (from -r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/99/55/6c020e22f81b2b76a1295b07ac92aa4f9aaf44dcb9cead89a6a24a3d828c/awscli-1.16.309-py2.py3-none-any.whl
Collecting rsa<=3.5.0,>=3.1.2 (from awscli->-r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/e1/ae/baedc9cb175552e95f3395c43055a6a5e125ae4d48a1d7a924baca83e92e/rsa-3.4.2-py2.py3-none-any.whl
Collecting botocore==1.13.45 (from awscli->-r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/96/22/9f8201d900956e57a9811e1b1c91c9f76c87487c76f636c2df1ce8379c38/botocore-1.13.45-py2.py3-none-any.whl
Collecting docutils<0.16,>=0.10 (from awscli->-r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/22/cd/a6aa959dca619918ccb55023b4cb151949c64d4d5d55b3f4ffd7eee0c6e8/docutils-0.15.2-py3-none-any.whl
Collecting colorama<0.4.2,>=0.2.5; python_version != "2.6" and python_version != "3.3" (from awscli->-r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/4f/a6/728666f39bfff1719fc94c481890b2106837da9318031f71a8424b662e12/colorama-0.4.1-py2.py3-none-any.whl
Collecting PyYAML<5.3,>=3.10; python_version != "2.6" and python_version != "3.3" (from awscli->-r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/8d/c9/e5be955a117a1ac548cdd31e37e8fd7b02ce987f9655f5c7563c656d5dcb/PyYAML-5.2.tar.gz
Collecting s3transfer<0.3.0,>=0.2.0 (from awscli->-r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/16/8a/1fc3dba0c4923c2a76e1ff0d52b305c44606da63f718d14d3231e21c51b0/s3transfer-0.2.1-py2.py3-none-any.whl
Collecting pyasn1>=0.1.3 (from rsa<=3.5.0,>=3.1.2->awscli->-r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/62/1e/a94a8d635fa3ce4cfc7f506003548d0a2447ae76fd5ca53932970fe3053f/pyasn1-0.4.8-py2.py3-none-any.whl
Collecting urllib3<1.26,>=1.20; python_version >= "3.4" (from botocore==1.13.45->awscli->-r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/b4/40/a9837291310ee1ccc242ceb6ebfd9eb21539649f193a7c8c86ba15b98539/urllib3-1.25.7-py2.py3-none-any.whl
Collecting jmespath<1.0.0,>=0.7.1 (from botocore==1.13.45->awscli->-r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/83/94/7179c3832a6d45b266ddb2aac329e101367fbdb11f425f13771d27f225bb/jmespath-0.9.4-py2.py3-none-any.whl
Collecting python-dateutil<3.0.0,>=2.1; python_version >= "2.7" (from botocore==1.13.45->awscli->-r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/d4/70/d60450c3dd48ef87586924207ae8907090de0b306af2bce5d134d78615cb/python_dateutil-2.8.1-py2.py3-none-any.whl
Collecting six>=1.5 (from python-dateutil<3.0.0,>=2.1; python_version >= "2.7"->botocore==1.13.45->awscli->-r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/65/26/32b8464df2a97e6dd1b656ed26b2c194606c16fe163c695a992b36c11cdf/six-1.13.0-py2.py3-none-any.whl
Building wheels for collected packages: PyYAML
  Running setup.py bdist_wheel for PyYAML ... error
  Complete output from command /home/romel/mercadochevere/venv/bin/python3.6 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-m00o5yhf/PyYAML/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmph_0_4ntgpip-wheel- --python-tag cp36:
  usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: -c --help [cmd1 cmd2 ...]
     or: -c --help-commands
     or: -c cmd --help
  
  error: invalid command 'bdist_wheel'
  
  ----------------------------------------
  Failed building wheel for PyYAML
  Running setup.py clean for PyYAML
Failed to build PyYAML
Installing collected packages: pyasn1, rsa, docutils, urllib3, jmespath, six, python-dateutil, botocore, colorama, PyYAML, s3transfer, awscli
  Running setup.py install for PyYAML ... done
Successfully installed PyYAML-5.2 awscli-1.16.309 botocore-1.13.45 colorama-0.4.1 docutils-0.15.2 jmespath-0.9.4 pyasn1-0.4.8 python-dateutil-2.8.1 rsa-3.4.2 s3transfer-0.2.1 six-1.13.0 urllib3-1.25.7
END: mié dic 25 10:28:10 -05 2019
```

requirements.txt
```
awscli
```

install.sh
```
# Time tracker Start
echo START: `date`

# Set defaults values
PROJECT_FOLDER="$(pwd)"
VENV_DIR=${PROJECT_FOLDER}/venv

# If exist Venv directory, delete
if [ -d "$VENV_DIR" ]; then
    rm -rf ./venv
fi

# Create venv directory
python3.6 -m venv ./venv

# Activate venv directory
. ./venv/bin/activate

# Install requirements
pip3.6 install -r requirements.txt

# export PYTHONPATH="${PYTHONPATH}:${PROJECT_FOLDER}/src"

# Time tracker End
echo END: `date`
```

**Comment by shlomi-viz** [comment]
I'm seen the same issue while try to install on a docker with `ubuntu` and trying to install `pyyaml` as a dependency of `keras==2.2.4`

```
  Running setup.py bdist_wheel for pyyaml: started
  Running setup.py bdist_wheel for pyyaml: finished with status 'error'
  Complete output from command /root/.virtualenvs/dev3/bin/python3.7 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-qex_v37e/pyyaml/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmp2_go5ypgpip-wheel- --python-tag cp37:
  usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: -c --help [cmd1 cmd2 ...]
     or: -c --help-commands
     or: -c cmd --help

  error: invalid command 'bdist_wheel'

  ----------------------------------------
  Failed building wheel for pyyaml
  Running setup.py clean for pyyaml
```

any ideas?

**Comment by kdaily** [comment]
The PRs that should have bumped PyYaml was merged: #4355 #4400

PyYaml will be bumped again very soon. Is this still an issue for anyone?
